src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorHtmlPublishDiagnosticsInterceptor.cs

```diff
@@ -0,0 +1,195 @@
+            }
+
+            if (documentManager is null)
+            {
+                throw new ArgumentNullException(nameof(documentManager));
+            }
+
+            if (documentMappingProvider is null)
+            {
+                throw new ArgumentNullException(nameof(documentMappingProvider));
+            }
+
+            if (documentSynchronizer is null)
+            {
+                throw new ArgumentNullException(nameof(documentSynchronizer));
+            }
+
+            _requestInvoker = requestInvoker;
+            _documentManager = documentManager;
+            _documentMappingProvider = documentMappingProvider;
+            _documentSynchronizer = documentSynchronizer;
+        }
+
+        public override async Task<InterceptionResult> ApplyChangesAsync(JToken token, string containedLanguageName, CancellationToken cancellationToken)
+        {
+            if (token is null)
+            {
+                throw new ArgumentNullException(nameof(token));
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            var diagnosticParams = token.ToObject<VSPublishDiagnosticParams>();
+
+            if (diagnosticParams is null)
+            {
+                throw new ArgumentException("Conversion of token failed.");
+            }
+
+            // We only support interception of Virtual HTML Files
+            if (!RazorLSPConventions.IsVirtualHtmlFile(diagnosticParams.Uri))
+            {
+                return CreateDefaultResponse(token);
+            }
+
+            var razorDocumentUri = RazorLSPConventions.GetRazorDocumentUri(diagnosticParams.Uri);
+
+            if (!_documentManager.TryGetDocument(razorDocumentUri, out var razorDocumentSnapshot))
+            {
+                return CreateDefaultResponse(token);
+            }
+
+            if (!razorDocumentSnapshot.TryGetVirtualDocument<HtmlVirtualDocumentSnapshot>(out var htmlDocumentSnapshot))
+            {
+                return CreateDefaultResponse(token);
+            }
+
+            // Ensure we're working with the virtual document specified in the diagnostic params
+            if (!htmlDocumentSnapshot.Uri.Equals(diagnosticParams.Uri))
+            {
+                return CreateDefaultResponse(token);
+            }
+
+            // Note; this is an `interceptor` & not a handler, hence 
+            // it's possible another interceptor mutates this request 
+            // later in the toolchain. Such an interceptor would likely 
+            // expect a `__virtual.html` suffix instead of `.razor`.
+            diagnosticParams.Uri = razorDocumentUri;
+
+            // Return early if there aren't any diagnostics to process
+            if (diagnosticParams.Diagnostics?.Any() != true)
+            {
+                return CreateResponse(diagnosticParams);
+            }
+
+            var synchronized = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync(
```

Comment:
Synchronizing isn't important here because technically we're already being told about the latest information from the sub-language servers. It's only important when you're making requests to one of the sub-langauge servers (not Razor since it's the all knowing being).

Reply:
Got it, thanks. Removed.